langchain/llms/custom_endpoint.py

```diff
@@ -0,0 +1,133 @@
+"""Wrapper around custom APIs."""
+from typing import Any, Dict, List, Mapping, Optional
+
+import requests
+import asyncio
+from functools import partial
+from pydantic import Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+from langchain.utils import get_from_dict_or_env
+
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class CustomEndpoint(LLM):
+    """Wrapper around custom Inference Endpoints, related to HuggingFaceHub Inference Endpoints.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import CustomEndpoint
+            endpoint_url = "https://api/endpoint/
+            llm = CustomEndpoint(
+                endpoint_url=endpoint_url,
+                api_key="my-api-key"
+            )
+    """
+
+    endpoint_url: str = ""
+    """Endpoint URL to use."""
+    task: Optional[str] = None
+    """Task to call the model with. Should be a task that returns `generated_text`."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    api_key: Optional[str] = None
```

Comment:
If this is truly custom, it may be better to let the user specify Headers: Optional[dict] = Field(default=None)`.

Since some endpoints may use keys like `X-Auth-MyToken: <token>`